clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -320,6 +323,19 @@ static bool isSYCLKernelBodyFunction(FunctionDecl *FD) {
   return FD->getOverloadedOperator() == OO_Call;
 }
 
+static bool isSYCLUndefinedAllowed(const FunctionDecl *Callee,
+                                   const SourceManager &SrcMgr) {
+  if (!Callee)
+    return false;
+  if (Callee->getName() == GlibcxxFailedAssertion) {
+    if (Callee->getLocation().printToString(SrcMgr).rfind(
```

Comment:
This part doesn't make me super comfortable because any path with bits/c++config.h will suffice, but I think this is enough of an edge case that we don't really *have* to do it. One question I have though: do we need to do the expensive check for the path to the header or can we just look to see if `Callee` is declared in a system header and call that good enough?

Reply:
Hi, @AaronBallman 
Checking whether "__failed_assertion" is declared in a system header should be good enough, do we have any elegant way to check whether a file path is a system path?(For our case, this function will be called in some libstdc++ header whose path is "/usr/include/c++/11/....").
Thanks very much.

Reply:
Yup, it's actually even easier than the existing code, which is a nice benefit: `SrcMgr->isInSystemHeader(Callee->getLocation())`